From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Sun, 27 Aug 2017 21:14:01 +0200
Subject: use icudt??{b,l}.dat depending on architecture

Upstream doesn't really care about big endian platforms and so also don't
care about a needed file icudt??b.dat on big endian platforms. Changing
the m4 macro for ICU so the needed file will be depended on the build
platform exported and let later the configure run automatically use the
correct name for the ICU *.dat file so we not only depend on the
prebuild *.dat file for little endian.

The `case foo in ...` loop could be obviously more compressed to only two
cases, one with architectures which are little endian and a second with all
big endian platforms. For more readability the more exclusive extra list
of all possible architectures in Debian was choosen.
---
 mozilla/build/autoconf/icu.m4 | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/mozilla/build/autoconf/icu.m4 b/mozilla/build/autoconf/icu.m4
index e0f922d..0252375 100644
--- a/mozilla/build/autoconf/icu.m4
+++ b/mozilla/build/autoconf/icu.m4
@@ -79,7 +79,31 @@ if test -n "$USE_ICU"; then
     # We could make this set as 'l' or 'b' for little or big, respectively,
     # but we'd need to check in a big-endian version of the file.
     if test -z "$ICU_DATA_FILE"; then
-        ICU_DATA_FILE="icudt${version}l.dat"
+        case $host in
+            dnl big endian architectures
+            hppa-*)
+                ICU_DATA_FILE="icudt${version}b.dat"
+                ;;
+            m68k-*)
+                ICU_DATA_FILE="icudt${version}b.dat"
+                ;;
+            mips-*)
+                ICU_DATA_FILE="icudt${version}b.dat"
+                ;;
+            s390*-*)
+                ICU_DATA_FILE="icudt${version}b.dat"
+                ;;
+            powerpc-*)
+                ICU_DATA_FILE="icudt${version}b.dat"
+                ;;
+            sparc*-*)
+                ICU_DATA_FILE="icudt${version}b.dat"
+                ;;
+            dnl assume little endian by default
+            *)
+                ICU_DATA_FILE="icudt${version}l.dat"
+                ;;
+        esac
     fi
 
     dnl We won't build ICU data as a separate file when building
